internal/syncutil/limitgroup.go

```diff
@@ -0,0 +1,67 @@
+/*
+Copyright The ORAS Authors.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package syncutil
+
+import (
+	"context"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// A LimitGroup is a collection of goroutines working on subtasks that are part of
+// the same overall task.
+type LimitGroup struct {
```

Comment:
Instead of `LimitGroup` / `NewLimitGroup()` pair, it is better to align the name to `LimitedGroup` / `LimitGroup()`, just like `io.LimitedReader` / `io.LimitReader()` and `LimitedRegion` / `LimitRegion`.

Reply:
I disagree with the change to the constructor's name, but I made the change as requested.

`LimitReader` is taking in a `Reader` and limiting it, so its name is like a command "limit this reader". I think `NewLimitedGroup` would be better than `LimitGroup`, since it's creating the `LimitedGroup` and not just wrapping an existing `Group` with limits.